utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/ConfluentUtils.scala

```diff
@@ -60,14 +62,37 @@ object ConfluentUtils extends LazyLogging {
       .valueOr(exc => throw new SerializationException(exc.getMessage, exc))
       .getInt
 
-  def serializeRecordToBytesArray(record: GenericRecord, schemaId: Int): Array[Byte] = {
-    val bos = new ByteArrayOutputStream()
-    writeSchemaId(schemaId, bos)
-    val encoder = EncoderFactory.get().binaryEncoder(bos, null)
-    val writer = new GenericDatumWriter[GenericRecord](record.getSchema, AvroUtils.genericData)
-    writer.write(record, encoder)
-    encoder.flush()
-    bos.toByteArray
+  /**
+    * Based on serializeImpl from [[io.confluent.kafka.serializers.AbstractKafkaAvroSerializer]]
+    */
+  def serializeDataToBytesArray(data: Any, schemaId: Int): Array[Byte] = {
```

Comment:
Please add migration guide about this change. I think that It could be used somewhere in client code